concordium-consensus/src/Concordium/GlobalState/Persistent/Account/StructureV1.hs

```diff
@@ -0,0 +1,1314 @@
+        !paedHash = getHash hashInputs
+    return $! PersistentAccountEnduringData{..}
+
+-- |[For internal use.] Recompute the Merkle hash of the enduring account data.
+rehashAccountEnduringData :: (MonadBlobStore m) => PersistentAccountEnduringData 'AccountV2 -> m (PersistentAccountEnduringData 'AccountV2)
+rehashAccountEnduringData ed = do
+    amhi2PersistingAccountDataHash <- getHashM (paedPersistingData ed)
+    (amhi2AccountStakeHash :: AccountStakeHash 'AccountV2) <- getHashM (paedStake ed)
+    amhi2EncryptedAmountHash <- case paedEncryptedAmount ed of
+        Null -> return initialAccountEncryptedAmountHash
+        Some e -> getHash <$> (loadPersistentAccountEncryptedAmount =<< refLoad e)
+    amhi2AccountReleaseScheduleHash <- case paedReleaseSchedule ed of
+        Null -> return Transient.emptyAccountReleaseScheduleHash
+        Some (rs, _) -> getHashM rs
+    let hashInputs :: AccountMerkleHashInputs 'AccountV2
+        hashInputs = AccountMerkleHashInputsV2{..}
+    return $! ed{paedHash = getHash hashInputs}
+
+enduringDataFlags :: PersistentAccountEnduringData av -> EnduringDataFlags
+enduringDataFlags PersistentAccountEnduringData{..} =
+    EnduringDataFlags
+        { edHasEncryptedAmount = isNotNull paedEncryptedAmount,
+          edHasReleaseSchedule = isNotNull paedReleaseSchedule,
+          edStakeFlags = stakeFlags paedStake
+        }
+
+-- * Enduring account data storage helper definitions
+
+-- |The nature of a pending stake change, abstracting the details.
+-- This is stored as the low-order 2 bits of a 'Word8'.
+data PendingChangeFlags
+    = PendingChangeNone -- ^No change is pending
+    | PendingChangeReduce -- ^A stake reduction is pending
+    | PendingChangeRemove -- ^Removal of stake is pending
+    deriving (Eq, Ord, Show)
+
+-- |Get the 'PendingChangeFlags' for a 'StakePendingChange''.
+stakePendingChangeFlags :: StakePendingChange' et -> PendingChangeFlags
+stakePendingChangeFlags NoChange = PendingChangeNone
+stakePendingChangeFlags ReduceStake{} = PendingChangeReduce
+stakePendingChangeFlags RemoveStake{} = PendingChangeRemove
+
+-- |Store a 'PendingChangeFlags' as the low-order 2 bits of a 'Word8'.
+pendingChangeFlagsToBits :: PendingChangeFlags -> Word8
+pendingChangeFlagsToBits PendingChangeNone = 0b00
+pendingChangeFlagsToBits PendingChangeReduce = 0b01
+pendingChangeFlagsToBits PendingChangeRemove = 0b10
+
+-- |Load a 'PendingChangeFlags' from the low-order 2 bits of a 'Word8'.
+-- All other bits must be 0.
+pendingChangeFlagsFromBits :: Word8 -> Either String PendingChangeFlags
+pendingChangeFlagsFromBits 0b00 = return PendingChangeNone
+pendingChangeFlagsFromBits 0b01 = return PendingChangeReduce
+pendingChangeFlagsFromBits 0b10 = return PendingChangeRemove
+pendingChangeFlagsFromBits _ = Left "Invalid pending change type"
+
+-- |Flags that represent the nature of the stake on an account.
+-- These are stored as the low-order 6 bits of a 'Word8'.
```

Comment:
This sentence does not really explain how they are stored and what those 6 bits mean.